src/providers/live/live-provider.ts

```diff
@@ -1,3 +1,211 @@
+    this._logger.error('Got an error during connection to push server');
+  };
+
+  private _fixCuePointEndTime<T extends ThumbPushNotificationData | SlideViewChangePushNotificationData>(cuePoints: T[]) {
+    return cuePoints
+      .sort((a, b) => {
+        return a.createdAt - b.createdAt;
+      })
+      .map((cue, index) => {
+        // fix endTime and replace VTTCue
+        if (cue.endTime === Number.MAX_SAFE_INTEGER && index !== cuePoints.length - 1) {
+          const fixedCue = {...cue, endTime: cuePoints[index + 1].startTime};
+          this._player.cuePointManager.addCuePoints([fixedCue]);
+        }
+        return cue;
+      });
+  }
+
+  private _prepareThumbCuePoints = (newThumb: ThumbPushNotificationData) => {
+    const startTime = this._player.currentTime - (this._currentTimeLive - newThumb.createdAt);
+    const newThumbCue = {
+      ...newThumb,
+      startTime: Number(startTime.toFixed(2)),
+      endTime: Number.MAX_SAFE_INTEGER,
+      assetUrl: makeAssetUrl(this._player.config.provider.env?.serviceUrl, newThumb.assetId, this._player.config.session?.ks)
+    };
+    this._thumbCuePoints.push(newThumbCue);
+    this._thumbCuePoints = this._fixCuePointEndTime(this._thumbCuePoints);
+    this._player.cuePointManager.addCuePoints([newThumbCue]);
+  };
+
+  private _prepareViewChangeCuePoints = (viewChange: SlideViewChangePushNotificationData) => {
+    const startTime = this._player.currentTime - (this._currentTimeLive - viewChange.createdAt);
+    try {
+      const partnerData = JSON.parse(viewChange.partnerData);
+      const newViewChangeCue = {
+        ...viewChange,
+        startTime: Number(startTime.toFixed(2)),
+        endTime: Number.MAX_SAFE_INTEGER,
+        partnerData
+      };
+      this._slideViewChangeCuePoints.push(newViewChangeCue);
+      this._slideViewChangeCuePoints = this._fixCuePointEndTime(this._slideViewChangeCuePoints);
+      this._player.cuePointManager.addCuePoints([newViewChangeCue]);
+    } catch (e) {
+      this._logger.error('Unnable parse slide-view change cue-point');
+    }
+  };
+
+  private _handleThumbNotificationData = ({thumbs}: ThumbNotificationsEvent) => {
+    this._currentTimeLivePromise.then(() => {
+      thumbs.forEach(thumb => this._prepareThumbCuePoints(thumb));
+    });
+  };
+
+  private _handleSlideViewChangeNotificationData = ({slideViewChanges}: SlideViewChangeNotificationsEvent) => {
+    this._currentTimeLivePromise.then(() => {
+      slideViewChanges.forEach(viewChange => this._prepareViewChangeCuePoints(viewChange));
+    });
+  };
+
+  // Placeholder for AOA cue-points
+  // private _handlePublicNotificationsData(data: PublicNotificationsEvent) {
+  //   console.log('>> handlePublicNotificationsData', data);
+  //   // TODO: prepare data and push to cue-point manager
+  // }
+
+  private _handlePushNotificationsErrorData(data: NotificationsErrorEvent) {
+    this._logger.error('Got an error from push notification server');
```

Comment:
Maybe we should warn and not error. See @OrenMe comment from other PR:
https://github.com/kaltura/playkit-js-kaltura-cuepoints/pull/2#discussion_r719486162

Reply:
And add the error to the printout